src/__tests__/calculate.test.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  render, screen, fireEvent, cleanup,
+} from '@testing-library/react';
+import calculate from '../logic/calculate';
+import CalculatorPage from '../route/calculator';
+
+afterEach(() => cleanup());
+
+describe(('Testing the calculator functions'), () => {
+  it('test add operation', () => {
+    render(<CalculatorPage />);
+    const one = screen.getByText('5');
+    const sum = screen.getByText('+');
+    const two = screen.getByText('7');
+    const equals = screen.getByText('=');
+    const result = screen.getByTestId('result');
+    fireEvent.click(one);
+    fireEvent.click(sum);
+    fireEvent.click(two);
+    fireEvent.click(equals);
+    expect(result.textContent).toBe('12');
+  });
+
+  test('test subtract operation', () => {
+    render(<CalculatorPage />);
+    const one = screen.getByText('5');
+    const sub = screen.getByText('-');
+    const two = screen.getByText('7');
+    const equals = screen.getByText('=');
+    const result = screen.getByTestId('result');
+    fireEvent.click(one);
+    fireEvent.click(sub);
+    fireEvent.click(two);
+    fireEvent.click(equals);
+    expect(result.textContent).toBe('-2');
+  });
+
+  it('test multiply operation', () => {
+    render(<CalculatorPage />);
+    const one = screen.getByText('5');
+    const divide = screen.getByText('÷');
+    const two = screen.getByText('5');
+    const equals = screen.getByText('=');
+    const result = screen.getByTestId('result');
+    fireEvent.click(one);
+    fireEvent.click(divide);
+    fireEvent.click(two);
+    fireEvent.click(equals);
+    expect(result.textContent).toBe('1');
+  });
```

Comment:
- [ ] Kindly always make sure that your `test` is passing before asking for the review. Your first three test cases of the `calculate.test.js` file are not passing and it seems like the errors are coming from the snippet that you're using in your test cases. Please fix your issues and re-submit your project with all passing tests.

```js 
const result = screen.getByTestId('result');
```

### Screenshot 1

![Screenshot from 2023-09-01 14-30-01](https://github.com/misikir21/mama/assets/66476329/075c6610-858f-4191-90c1-f5f13a3cc745)

### Screenshot 2

![Screenshot from 2023-09-01 14-30-10](https://github.com/misikir21/mama/assets/66476329/4da33b92-403a-4bdf-9abe-ab716b673f28)

### Screenshot 3

![Screenshot from 2023-09-01 14-30-30](https://github.com/misikir21/mama/assets/66476329/d567b993-68c4-41fb-a5a0-5638a79718bb)